### DISCUSSION TITLE
Is it possible to create a UI that only consumes part of the terminal's height (similar to fzf)

### DISCUSSION THREAD

**Discussion by dob9601** [gyscos/cursive#732] [Q&A]
Title basically says it all, i was wondering if it were possible to use this framework to create a UI that only uses part of the terminal's height - as shown in the screenshot below:
![image](https://github.com/gyscos/cursive/assets/24723950/51d845b8-30ea-4749-a26d-54988abffe16)
Interestingly, fzf only seems to support the behaviour in certain shells

**Comment by gyscos** [comment]
Hi, and thanks for the question!

That's not currently supported. While it's possible to have a layout that only uses part of the height, it still won't show the previous content of the terminal (we use the "alternate terminal" concept, which hides any previous content while cursive is running, and hides any cursive content on exit).

The closest you could get is a full-screen layout that includes the "CLI" part, and adds things underneath when needed.
